### ISSUE TITLE
Support a way to reason about a field's type

### ISSUE THREAD

**Issue by bradcray** [chapel-lang/chapel#26444]
This issue requests ways to reason about the type of a record field, such as:

```chapel
…myRecord.myField.type…
```

Note that this is somewhat related to issue #11816 (which is similar, but more general) and https://github.com/chapel-lang/chapel/issues/12167 which is attractive due to the ability to get a field's type without knowing its name.

**Comment by DanilaFe** [comment]
I was a little confused at first, but I think you're requesting the following:

```Chapel
record R {
  var x: int;
}
type t = R.x.type;
```

Initially, I interpreted your request as:

```Chapel
record R {
  var x: int;
}
var myRecord: R;
type t = myRecord.x.type;
```

Which already works today. 

Just commenting this for anyone else who might get confused like I did.

**Comment by bradcray** [comment]
Good point Daniel—I've updated the title and OP to try and clarify this.
